lib/shim/shim.js

```diff
@@ -1313,7 +1313,7 @@ function applySegment(func, segment, full, context, args, inContextCB) {
       return fnApply.call(func, context, args)
```

Comment:
this code path could be optimized like in transaction/tracer as i don't see it getting used.

```js
  return contextManager.runInContext(segment, () => {
    if (full) {
      segment.start()
    }

    if (typeof inContextCB === 'function') {
      inContextCB(segment)
    }

    try {
      return fnApply.call(func, context, args)
    } finally {
      if (full) {
        segment.touch()
      }
    }
  })
```

Reply:
take note I removed the call to get the previous segment as it was only used to add it on the domain that's never being used

Reply:
I think this is the same case as what we saw in `transaction/tracer`, where the segment is being set and used later in `shim.getSegment(this)`. So it is being used, in case someone is still using deprecated domains.